[Fastlane.Swift] FastlaneSwiftRunner fails to build

### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

Since the update from `2.183.2` to `2.184.0` the FastlaneSwiftRunner fails to build.

`the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions` on several occasions in `Fastlane.swift`

Fastlane.Swift was intialized from a clean state and nothing custom was added to make sure no code was affecting it.

##### Command executed

- `bundle exec fastlane init swift`
- `bundle exec fastlane custom`

##### Complete output when running fastlane, including the stack trace and command used

<details><pre>

```
[✔] 🚀 
[18:39:15]: loading manifest: /Users/david/.gem/ruby/2.7.0/gems/fastlane-2.184.0/fastlane/swift/upgrade_manifest.json
[18:39:15]: FastlaneSwiftRunner project is up-to-date
[18:39:15]: Resolving Swift Package Manager dependencies...
[18:39:15]: $ xcodebuild -resolvePackageDependencies -scheme FastlaneRunner -project ./fastlane/swift/FastlaneSwiftRunner/FastlaneSwiftRunner.xcodeproj
[18:39:16]: ▸ Command line invocation:
[18:39:16]: ▸     /Applications/Xcode-12.5.0.app/Contents/Developer/usr/bin/xcodebuild -resolvePackageDependencies -scheme FastlaneRunner -project ./fastlane/swift/FastlaneSwiftRunner/FastlaneSwiftRunner.xcodeproj
[18:39:16]: ▸ User defaults from command line:
[18:39:16]: ▸     IDEPackageSupportUseBuiltinSCM = YES
[18:39:16]: ▸ resolved source packages: 
[18:39:16]: $ xcodebuild -showBuildSettings -scheme FastlaneRunner -project ./fastlane/swift/FastlaneSwiftRunner/FastlaneSwiftRunner.xcodeproj
[18:39:17]: $ set -o pipefail && xcodebuild -scheme FastlaneRunner -project ./fastlane/swift/FastlaneSwiftRunner/FastlaneSwiftRunner.xcodeproj -destination 'generic/platform=macOS' build | tee /Users/david/Library/Logs/gym/App-FastlaneRunner.log | xcpretty
▸ Compiling RubyCommand.swift
▸ Compiling RubyCommandable.swift
▸ Compiling Gymfile.swift
▸ Compiling MatchfileProtocol.swift
▸ Compiling SocketClientDelegateProtocol.swift
▸ Compiling LaneFileProtocol.swift
▸ Compiling Precheckfile.swift
▸ Compiling ScanfileProtocol.swift
▸ Compiling Matchfile.swift
▸ Compiling ScreengrabfileProtocol.swift
▸ Compiling ArgumentProcessor.swift
▸ Compiling GymfileProtocol.swift
▸ Compiling SnapshotfileProtocol.swift
▸ Compiling Deliverfile.swift
▸ Compiling Fastfile.swift
▸ Compiling Actions.swift
▸ Compiling Appfile.swift
▸ Compiling SocketClient.swift
▸ Compiling DeliverfileProtocol.swift
▸ Compiling Runner.swift
▸ Compiling RunnerArgument.swift
▸ Compiling ControlCommand.swift
▸ Compiling main.swift
▸ Compiling PrecheckfileProtocol.swift
▸ Compiling SocketResponse.swift
▸ Compiling Scanfile.swift
▸ Compiling Plugins.swift
▸ Compiling Snapshotfile.swift
▸ Compiling Fastlane.swift
▸ Compiling Screengrabfile.swift
▸ Compiling OptionalConfigValue.swift
▸ Compiling RubyCommand.swift
▸ Compiling RubyCommandable.swift
▸ Compiling Gymfile.swift
▸ Compiling MatchfileProtocol.swift
▸ Compiling SocketClientDelegateProtocol.swift
▸ Compiling LaneFileProtocol.swift
▸ Compiling Precheckfile.swift
▸ Compiling ScanfileProtocol.swift
▸ Compiling Matchfile.swift
▸ Compiling ScreengrabfileProtocol.swift
▸ Compiling ArgumentProcessor.swift
▸ Compiling GymfileProtocol.swift
▸ Compiling SnapshotfileProtocol.swift
▸ Compiling Deliverfile.swift
▸ Compiling Fastfile.swift
▸ Compiling Actions.swift
▸ Compiling Appfile.swift
▸ Compiling SocketClient.swift
▸ Compiling DeliverfileProtocol.swift
▸ Compiling Runner.swift
** BUILD FAILED **


The following build commands failed:
	CompileSwift normal x86_64 /Users/.../fastlane/swift/Plugins.swift
	CompileSwift normal x86_64 /Users/.../fastlane/swift/Snapshotfile.swift
	CompileSwift normal x86_64 /Users/.../fastlane/swift/Fastlane.swift
	CompileSwift normal x86_64 /Users/.../fastlane/swift/Screengrabfile.swift
	CompileSwift normal x86_64 /Users/.../fastlane/swift/OptionalConfigValue.swift
	CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler
	CompileSwift normal arm64 /Users/.../fastlane/swift/Fastlane.swift
(7 failures)
▸ Compiling RunnerArgument.swift
▸ Compiling ControlCommand.swift
▸ Compiling main.swift
▸ Compiling PrecheckfileProtocol.swift
▸ Compiling SocketResponse.swift
▸ Compiling Scanfile.swift
▸ Compiling Plugins.swift
▸ Compiling Snapshotfile.swift
▸ Compiling Fastlane.swift

❌  /Users/.../fastlane/swift/Fastlane.swift:625:10: the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions

        .compactMap { $0 }
~~~~~~~~~^~~~~~~~~~~~~~~~~



❌  /Users/.../fastlane/swift/Fastlane.swift:892:10: the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions

        .compactMap { $0 }
~~~~~~~~~^~~~~~~~~~~~~~~~~



❌  /Users/.../fastlane/swift/Fastlane.swift:1453:10: the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions

        .compactMap { $0 }
~~~~~~~~~^~~~~~~~~~~~~~~~~



❌  /Users/.../fastlane/swift/Fastlane.swift:1645:10: the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions

        .compactMap { $0 }
~~~~~~~~~^~~~~~~~~~~~~~~~~



❌  /Users/.../fastlane/swift/Fastlane.swift:1841:10: the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions

        .compactMap { $0 }
~~~~~~~~~^~~~~~~~~~~~~~~~~



❌  /Users/.../fastlane/swift/Fastlane.swift:2229:10: the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions

        .compactMap { $0 }
~~~~~~~~~^~~~~~~~~~~~~~~~~



❌  /Users/.../fastlane/swift/Fastlane.swift:2433:10: the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions

        .compactMap { $0 }
~~~~~~~~~^~~~~~~~~~~~~~~~~



❌  /Users/.../fastlane/swift/Fastlane.swift:2537:10: the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions

        .compactMap { $0 }
~~~~~~~~~^~~~~~~~~~~~~~~~~



❌  /Users/.../fastlane/swift/Fastlane.swift:3835:10: the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions

        .compactMap { $0 }
~~~~~~~~~^~~~~~~~~~~~~~~~~



❌  /Users/.../fastlane/swift/Fastlane.swift:5561:10: the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions

        .compactMap { $0 }
~~~~~~~~~^~~~~~~~~~~~~~~~~



❌  /Users/.../fastlane/swift/Fastlane.swift:5834:10: the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions

        .compactMap { $0 }
~~~~~~~~~^~~~~~~~~~~~~~~~~



❌  /Users/.../fastlane/swift/Fastlane.swift:6612:10: the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions

        .compactMap { $0 }
~~~~~~~~~^~~~~~~~~~~~~~~~~



❌  /Users/.../fastlane/swift/Fastlane.swift:6806:10: the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions

        .compactMap { $0 }
~~~~~~~~~^~~~~~~~~~~~~~~~~



❌  /Users/.../fastlane/swift/Fastlane.swift:7458:10: the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions

        .compactMap { $0 }
~~~~~~~~~^~~~~~~~~~~~~~~~~



❌  /Users/.../fastlane/swift/Fastlane.swift:8600:10: the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions

        .compactMap { $0 }
~~~~~~~~~^~~~~~~~~~~~~~~~~



❌  /Users/.../fastlane/swift/Fastlane.swift:8680:10: the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions

        .compactMap { $0 }
~~~~~~~~~^~~~~~~~~~~~~~~~~



❌  /Users/.../fastlane/swift/Fastlane.swift:8998:10: the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions

        .compactMap { $0 }
~~~~~~~~~^~~~~~~~~~~~~~~~~



❌  /Users/.../fastlane/swift/Fastlane.swift:9869:10: the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions

        .compactMap { $0 }
~~~~~~~~~^~~~~~~~~~~~~~~~~



❌  /Users/.../fastlane/swift/Fastlane.swift:10073:10: the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions

        .compactMap { $0 }
~~~~~~~~~^~~~~~~~~~~~~~~~~



❌  /Users/.../fastlane/swift/Fastlane.swift:10154:10: the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions

        .compactMap { $0 }
~~~~~~~~~^~~~~~~~~~~~~~~~~



❌  /Users/.../fastlane/swift/Fastlane.swift:10222:10: the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions

        .compactMap { $0 }
~~~~~~~~~^~~~~~~~~~~~~~~~~



❌  /Users/.../fastlane/swift/Fastlane.swift:10566:10: the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions

        .compactMap { $0 }
~~~~~~~~~^~~~~~~~~~~~~~~~~



❌  /Users/.../fastlane/swift/Fastlane.swift:10824:10: the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions

        .compactMap { $0 }
~~~~~~~~~^~~~~~~~~~~~~~~~~



❌  /Users/.../fastlane/swift/Fastlane.swift:11071:10: the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions

        .compactMap { $0 }
~~~~~~~~~^~~~~~~~~~~~~~~~~



❌  /Users/.../fastlane/swift/Fastlane.swift:11999:10: the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions

        .compactMap { $0 }
~~~~~~~~~^~~~~~~~~~~~~~~~~



❌  /Users/.../fastlane/swift/Fastlane.swift:12157:10: the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions

        .compactMap { $0 }
~~~~~~~~~^~~~~~~~~~~~~~~~~



❌  /Users/.../fastlane/swift/Fastlane.swift:12372:10: the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions

        .compactMap { $0 }
~~~~~~~~~^~~~~~~~~~~~~~~~~



❌  /Users/.../fastlane/swift/Fastlane.swift:12800:10: the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions

        .compactMap { $0 }
~~~~~~~~~^~~~~~~~~~~~~~~~~


▸ Compiling Screengrabfile.swift
▸ Compiling OptionalConfigValue.swift
[18:43:59]: Exit status: 65
+-----------+--------+
|    Lane Context    |
+-----------+--------+
| LANE_NAME | custom |
+-----------+--------+
[18:43:59]: Exit status: 65
[18:43:59]: fastlane finished with errors

[!] Exit status: 65
```

</pre></details>

### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                              |
| --------------------------- | -------------------------------------------------- |
| OS                          | 11.2.3                                             |
| Ruby                        | 2.7.3                                              |
| Bundler?                    | true                                               |
| Git                         | git version 2.31.1                                 |
| Installation Source         | ~/.gem/ruby/2.7.0/bin/fastlane                     |
| Host                        | macOS 11.2.3 (20D91)                               |
| Ruby Lib Dir                | /usr/local/Cellar/ruby@2.7/2.7.3/lib               |
| OpenSSL Version             | OpenSSL 1.1.1k  25 Mar 2021                        |
| Is contained                | false                                              |
| Is homebrew                 | false                                              |
| Is installed via Fabric.app | false                                              |
| Xcode Path                  | /Applications/Xcode-12.5.0.app/Contents/Developer/ |
| Xcode Version               | 12.5                                               |


### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |


### fastlane files:

<details><summary>`./fastlane/Fastfile.swift`</summary>

```ruby
// This file contains the fastlane.tools configuration
// You can find the documentation at https://docs.fastlane.tools
//
// For a list of all available actions, check out
//
//     https://docs.fastlane.tools/actions
//

import Foundation

class Fastfile: LaneFile {
	func customLane() {
	desc("Description of what the lane does")
		// add actions here: https://docs.fastlane.tools/actions
	}
}

```
</details>

**No Appfile found**


### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.184.0 | ✅ Up-To-Date  |


### Loaded fastlane plugins:

**No plugins Loaded**


<details><summary><b>Loaded gems</b></summary>

| Gem                             | Version      |
| ------------------------------- | ------------ |
| did_you_mean                    | 1.4.0        |
| bundler                         | 2.2.15       |
| uri                             | 0.10.0       |
| rake                            | 13.0.3       |
| CFPropertyList                  | 3.0.3        |
| public_suffix                   | 4.0.6        |
| addressable                     | 2.7.0        |
| artifactory                     | 3.0.15       |
| atomos                          | 0.1.3        |
| aws-eventstream                 | 1.1.1        |
| aws-partitions                  | 1.461.0      |
| aws-sigv4                       | 1.2.3        |
| jmespath                        | 1.4.0        |
| aws-sdk-core                    | 3.114.0      |
| aws-sdk-kms                     | 1.43.0       |
| aws-sdk-s3                      | 1.95.0       |
| babosa                          | 1.0.4        |
| claide                          | 1.0.3        |
| colored                         | 1.2          |
| colored2                        | 3.1.2        |
| highline                        | 2.0.3        |
| commander                       | 4.6.0        |
| declarative                     | 0.0.20       |
| digest-crc                      | 0.6.3        |
| unf_ext                         | 0.0.7.7      |
| unf                             | 0.1.4        |
| domain_name                     | 0.5.20190701 |
| dotenv                          | 2.7.6        |
| emoji_regex                     | 3.2.2        |
| excon                           | 0.81.0       |
| faraday-em_http                 | 1.0.0        |
| faraday-em_synchrony            | 1.0.0        |
| faraday-excon                   | 1.1.0        |
| faraday-net_http                | 1.0.1        |
| faraday-net_http_persistent     | 1.1.0        |
| multipart-post                  | 2.0.0        |
| ruby2_keywords                  | 0.0.4        |
| faraday                         | 1.4.2        |
| http-cookie                     | 1.0.3        |
| faraday-cookie_jar              | 0.0.7        |
| faraday_middleware              | 1.0.0        |
| fastimage                       | 2.2.3        |
| gh_inspector                    | 1.1.3        |
| jwt                             | 2.2.3        |
| memoist                         | 0.16.2       |
| multi_json                      | 1.15.0       |
| os                              | 1.1.1        |
| signet                          | 0.15.0       |
| googleauth                      | 0.16.2       |
| httpclient                      | 2.8.3        |
| mini_mime                       | 1.1.0        |
| trailblazer-option              | 0.1.1        |
| uber                            | 0.1.0        |
| representable                   | 3.1.1        |
| retriable                       | 3.1.2        |
| rexml                           | 3.2.5        |
| webrick                         | 1.7.0        |
| google-apis-core                | 0.3.0        |
| google-apis-androidpublisher_v3 | 0.3.0        |
| google-apis-playcustomapp_v1    | 0.2.0        |
| google-apis-iamcredentials_v1   | 0.3.0        |
| google-apis-storage_v1          | 0.3.0        |
| google-cloud-env                | 1.5.0        |
| google-cloud-errors             | 1.1.0        |
| google-cloud-core               | 1.6.0        |
| google-cloud-storage            | 1.31.1       |
| json                            | 2.5.1        |
| mini_magick                     | 4.11.0       |
| naturally                       | 2.2.1        |
| plist                           | 3.6.0        |
| rubyzip                         | 2.3.0        |
| security                        | 0.1.3        |
| simctl                          | 1.6.8        |
| terminal-notifier               | 2.0.0        |
| unicode-display_width           | 1.7.0        |
| terminal-table                  | 1.8.0        |
| tty-screen                      | 0.8.1        |
| tty-cursor                      | 0.7.1        |
| tty-spinner                     | 0.9.3        |
| word_wrap                       | 1.0.0        |
| nanaimo                         | 0.3.0        |
| xcodeproj                       | 1.19.0       |
| rouge                           | 2.0.7        |
| xcpretty                        | 0.3.0        |
| xcpretty-travis-formatter       | 1.0.1        |
</details>


*generated on:* **2021-05-22**
</details>

--- a/fastlane/lib/fastlane/swift_fastlane_function.rb
+++ b/fastlane/lib/fastlane/swift_fastlane_function.rb
@@ -325,7 +325,8 @@ module Fastlane
       if args.empty?
         implm += "let args: [RubyCommand.Argument] = []\n"
       else
-        implm += "let args = [#{args.group_by { |h| h[:name] }.keys.join(",\n")}]\n"
+        implm += "let array: [RubyCommand.Argument?] = [#{args.group_by { |h| h[:name] }.keys.join(",\n")}]\n"
+        implm += "let args: [RubyCommand.Argument] = array\n"
         implm += ".filter { $0?.value != nil }\n"
         implm += ".compactMap { $0 }\n"
       end

